multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/signature/SignatureKeyManager.java

```diff
@@ -57,56 +60,79 @@
 
   private final String algorithm;
   private final SignatureKeyDao signatureKeyDao;
+  private final EventService eventService;
+  private final EventSubscriber<?> workspaceEventsSubscriber;
 
   @Inject
   @SuppressWarnings("unused")
   private DBInitializer dbInitializer;
 
-  private KeyPair cachedPair;
+  private Map<String, KeyPair> cachedPair = new HashMap<>();
```

Comment:
maybe we should use guava loader cache here?

Reply:
We didn't use any cache-specific features yet (like auto expiring or auto eviction). The only useful thing is sychronization. So maybe we consider using `ConcurrentHashMap` here ?